custom_components/google_home/sensor.py

```diff
@@ -29,96 +18,107 @@
-                        ),
-                        GoogleHomeTokenSensor(
-                            coordinator,
-                            entry,
-                            device.device_name,
-                            device.local_auth_token,
-                        ),
-                    ]
-                )
-            else:
-                _LOGGER.warning(
-                    (
-                        "The %s device(hardware='%s') is not Google Home "
-                        "compatible and has no alarms/timers"
-                    ),
-                    device.device_name,
-                    device.hardware,
-                )
+        sensors.append(
+            GoogleHomeDeviceSensor(
+                coordinator,
+                entry,
+                device.name,
+                device.auth_token,
+            )
+        )
+        if device.auth_token and device.available:
+            sensors += [
+                GoogleHomeAlarmSensor(
+                    coordinator,
+                    entry,
+                    device.name,
+                ),
+                GoogleHomeNextAlarmSensor(
+                    coordinator,
+                    entry,
+                    device.name,
+                ),
+                GoogleHomeTimerSensor(
+                    coordinator,
+                    entry,
+                    device.name,
+                ),
+                GoogleHomeNextTimerSensor(
+                    coordinator,
+                    entry,
+                    device.name,
+                ),
+            ]
+    async_add_devices(sensors)
 
 
 class GoogleHomeSensorMixin:
     def get_device(self):
-        """Return the device matched by name
+        """Return the device matched by device name
         from the list of google devices in coordinator_data"""
         return next(
             (
                 device
                 for device in self.coordinator.data
-                if device.device_name == self._name
+                if device.name == self.device_name
             ),
             None,
         )
 
+    @staticmethod
+    def as_dict(obj_list):
+        """Return list of objects represented as dictionaries """
+        return [obj.__dict__ for obj in obj_list]
+
 
-class GoogleHomeTokenSensor(GoogleHomeSensorMixin, GoogleHomeTokenEntity):
-    """GoogleHome Sensor class."""
+class GoogleHomeDeviceSensor(GoogleHomeSensorMixin, GoogleHomeDeviceEntity):
+    """GoogleHome Device Sensor class."""
 
-    def __init__(self, coordinator, entry, device_name, local_auth_token):
+    def __init__(self, coordinator, entry, device_name, auth_token):
         """Initialize the sensor."""
         super().__init__(coordinator, entry)
-        self._name = device_name
+        self.device_name = device_name
+        self.auth_token = auth_token
+        device = self.get_device()
+        self.initial_attributes = self.get_device_attributes(device)
```

Comment:
Why do you need this? Can we return something like this if device is not available?
```
{
    "device_name": self.device_name,
    "auth_token": None,
    "ip_address": None,
    "hardware": None,
    "available": False,
    "integration": DOMAIN,
}
 ```

Reply:
haha, valid point! sorry i m getting a bit exhausted here :D